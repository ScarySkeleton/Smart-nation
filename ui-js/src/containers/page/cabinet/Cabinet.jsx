import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Cabinet extends PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className='container cabinet'>
                Cabinet
            </div>
        )
    }
};

Cabinet.propTypes = {

};

Cabinet.defultProps = {

};

export default Cabinet;