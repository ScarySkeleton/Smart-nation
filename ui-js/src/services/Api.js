const API_PATH = '/';

function statusChecker(response) {
    if(response.status >= 200 && response.status < 300)
        return Promise.resolve(response);
    return Promise.reject(response);
}

/*
    ================================================================
                    USER LOGIN\LOGOUT\REGESTRATION
    ================================================================
*/
export function loginRequest(userData) {
    return function () {
        return fetch("http://localhost:50363/Account/Login", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(userData)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            return {
                username: json.login
                , name: json.name
                , surname: json.surname
                , role: json.role
        }});
    }
};

export function registrationRequest(userData) {
    return function() {
        return fetch("http://localhost:50363/Account/Register", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(userData)
        })
        .then(statusChecker);
    }
}

export function logoutRequest() {
    return function() {
        return fetch("http://localhost:50363/Account/Logout", {
            method: "POST", 
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
        });
    }
}

/*
    ================================================
                    USER CABINET
    ================================================
*/
export function getCabinetData(data) {
    return function() {
        return {
            userId: 1,
            photo: null,
            description: 'My name is user and I like to read',
        }
    }
}

export function addBook(data) {
    return function() {
        return fetch("http://localhost:50363/PersonalCabinet/AddBook", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(data)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            console.log(json);
        })
    }
}

export function getBookShelfBooks() {
    return function() {
        return fetch("http://localhost:50363/PersonalCabinet/GetBook", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include"
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            console.log(json);
            return json;
        })
    }
}
