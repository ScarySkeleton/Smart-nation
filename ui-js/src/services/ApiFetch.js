const API_PATH='http://localhost:50363';

function statusChecker(response) {
    if(response.status >= 200 && response.status < 300)
        return Promise.resolve(response);
    return Promise.reject(response);
}

export function apiAuthPostFetch(path, data) {
    return function() {
        return fetch(`${API_PATH}/${path}`, {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(data)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            return json;
        });
    }
}

export function apiNonAuthPostFetch(path, data) {
    return function() {
        return fetch(`${API_PATH}/${path}`, {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            body: JSON.stringify(data)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            return json;
        });
    }
}