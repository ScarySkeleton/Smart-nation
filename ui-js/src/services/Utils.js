import Adapter from 'enzyme-adapter-react-15';
import { configure } from 'enzyme';

import valid from './validate/validate';

export function enzymeConf() {
    return configure({ adapter: new Adapter() });
}

export function validate(data, config) {
    valid.config = config;
    return valid.validate(data);
}
