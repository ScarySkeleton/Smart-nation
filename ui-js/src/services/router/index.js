import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import throttle from 'lodash/throttle';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

import { loadState, saveState } from '../localStorage/localStorage';

const savedState = loadState();
const sagaMiddleware = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;    

const store = createStore(
    rootReducer
    , savedState
    , composeEnhancers(applyMiddleware(sagaMiddleware)));

store.subscribe(throttle(() => {
    saveState({
        Login: store.getState().Login,
    })
}, 1000));
sagaMiddleware.run(rootSaga);

export default store;
