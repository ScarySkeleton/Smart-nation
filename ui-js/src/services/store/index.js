import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import throttle from 'lodash/throttle';
import thunk from 'redux-thunk';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

import { loadState, saveState } from '../localStorage/localStorage';

const savedState = loadState();
const sagaMiddleware = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;    

const store = createStore(
    rootReducer
    , savedState
    , composeEnhancers(applyMiddleware(sagaMiddleware)
    , applyMiddleware(thunk)));

store.subscribe(throttle(() => {
    saveState({
        Login: store.getState().Login,
    })
}, 1000));
sagaMiddleware.run(rootSaga);

export default store;
