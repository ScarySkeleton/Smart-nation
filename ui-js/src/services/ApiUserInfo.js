import {Error} from '../components/popup/Error/Error';
import {popupOpen} from '../components/popup/popup.action';
import store from './store/index';
import React from 'react';
import {connect} from 'react-redux';

/*
    ================================================
                    USER INFO
    ================================================
*/
const errorTitleMessage = 'Error was occured.'
const errorBodyMessage = `Error was occured while
    fetching data from server.`;



let fetchingError = (error) => {
    const errorPopup = {
        title: errorTitleMessage,
        body: <Error message={`${errorBodyMessage}
        Error message: ${error}`} /> 
    }
    
    return store.dispatch(popupOpen(errorPopup));
}


export {fetchingError};
