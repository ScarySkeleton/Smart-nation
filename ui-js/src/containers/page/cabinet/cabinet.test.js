import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { enzymeConf } from '../../../services/Utils';
import Cabinet from './Cabinet';

describe('Cabinet render', () => {
    beforeAll(() => {
        enzymeConf();
    });
    
    it('should render correctly', () => {
        const tree = shallow(<Cabinet />);
        expect(toJson(tree)).toMatchSnapshot();
    })
});