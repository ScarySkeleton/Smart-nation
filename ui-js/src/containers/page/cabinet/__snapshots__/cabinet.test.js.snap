// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cabinet render should render correctly 1`] = `
<div
  className="container cabinet"
>
  Cabinet
</div>
`;
